src/hazelweb/Model.re

```diff
@@ -86,10 +86,12 @@ type result_state =
   | ResultsDisabled
   | Result(has_result_state);
 
+[@deriving sexp]
+type history = ZList.t(Statics.edit_state, Statics.edit_state);
```

Comment:
Make a module `UndoHistory` where this is `type t`, and move the functions below into it.